### PR TITLE
Resolving on closed cache

### PR DESCRIPTION
**Pull request by aphelionz**
This PR:

1. Implements `open` and `close` functions that return if the database is already `open` or `closed` respectively, or perform the action
2. Added `this.handlers`
3. Added `Cache.removeHandlers`, to be called from `OrbitDB._onClose`, which is a hook from `Store.close` :nerd_face: 

Note: I didn't implement `addHandler` since there's no additional logic besides the native `this.handlers.add` call required yet.

### CODE REVIEW

File: src/Cache.js [hunk 1]
```diff
@@ -29,6 +48,8 @@ class Cache {
   // Set value in the cache and return the new value
   set (key, value) {
     return new Promise((resolve, reject) => {
+      if(this.status !== "open") resolve();
```
Comment (shamb0t) [inline src/Cache.js]: Should this reject since it doesnt set the value?

File: src/Cache.js [hunk 2]
```diff
@@ -7,12 +7,31 @@ Logger.setLogLevel('ERROR')
 class Cache {
   constructor (store) {
     this._store = store
+    this.handlers = new Set();
```
Comment (shamb0t) [inline src/Cache.js]: Can you explain this a bit? I don't see it used again

Comment (aphelionz) [reply src/Cache.js]: Yes, it's a set of store addresses that is managed by `orbit-db` when creating/opening/closing caches. It helps keep track of which db is using which cache.

Comment (shamb0t) [reply src/Cache.js]: Can some interface be added here to manipulate this property? It should be clear what its purpose is just from looking at this file

Comment (aphelionz) [reply src/Cache.js]: Yeah, that's a good call. I can create an `addHandler` function that does a lot of what `OrbitDb._onClose` does

File: src/Cache.js [hunk 3]
```diff
@@ -7,9 +7,26 @@ Logger.setLogLevel('ERROR')
 class Cache {
   constructor (store) {
     this._store = store
+    this.handlers = new Set()
```
[the remark this answers is not in the document]
Comment (aphelionz) [reply src/Cache.js]: Resolving the above and moving the conversation here. I will implemenet `async Cache.addHandler`

[the remark this answers is not in the document]
Comment (shamb0t) [reply src/Cache.js]: I think `addHandler` is useful too, believe it is also called in orbitdb

[the remark this answers is not in the document]
Comment (aphelionz) [reply src/Cache.js]: `store.handlers.add` is called, but not `addHandler`

[the remark this answers is not in the document]
Comment (aphelionz) [reply src/Cache.js]: Now I'm wondering whose job it is to tell Cache to add a handler when a Store gets/creates a cache...

Could it be Store? like `Cache.requestCache` or some such call, which would then in turn update the `handler` set? Or could we maybe emit an event that Cache listens for to update the stores and handler?

[the remark this answers is not in the document]
Comment (aphelionz) [reply src/Cache.js]: TBD

[the remark this answers is not in the document]
Comment (shamb0t) [reply src/Cache.js]: Yes I meant that orbtidb should call cache.addHandler instead (or perhaps Store should as you say)

**Pull request merged**